tests/qasm3/test_loop.py

```diff
@@ -332,3 +332,31 @@ def test_convert_qasm3_for_loop_unsupported_type(caplog):
 
     assert "Error at line 9, column 16" in caplog.text
     assert 'for bit b in "001"' in caplog.text
+
+
+def test_while_loop_with_break_and_continue():
```

Comment:
Can you add more tests for the change eg. - 
1. Loop limit exceeded exceptions 
2. Comparison of exact qasm strings for unrolled while loops
3. Empty while loops are ignored